### PR TITLE
PMM-8269 MongoDB Point-in-Time backups

### PR DESCRIPTION
**Pull request by artemgavrilov**
[PMM-8269](https://jira.percona.com/browse/PMM-8269)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/2001
---
- [ ] https://github.com/percona/pmm-managed/pull/811
- [ ] https://github.com/percona/pmm-agent/pull/285
---

### CODE REVIEW

File: api/managementpb/backup/backups.proto [hunk 1]
```diff
@@ -176,6 +187,17 @@ message RemoveScheduledBackupRequest {
 
 message RemoveScheduledBackupResponse {}
 
+message GetMongoPITRRangesRequest {}
```
Comment (Dasio) [inline api/managementpb/backup/backups.proto:190]: shouldn't be there something like service id?

File: api/agentpb/agent.proto [hunk 2]
```diff
@@ -283,6 +283,24 @@ message StartActionRequest {
     // Contains files and their contents which can be used in DSN.
     TextFiles text_files = 2;
   }
+  // PBMStatusParams describes pbm status action parameters.
+  message PBMStatusParams {
+    // DSN for the service. May contain connection (dial) timeout.
+    // May contain placeholders for file paths in DSN.
+    string dsn = 1;
+    // Contains files and their contents which can be used in DSN.
+    TextFiles text_files = 2;
```
Comment (Dasio) [inline api/agentpb/agent.proto:292]: what is this? why we need some textfiles?

Comment (artemgavrilov) [reply api/agentpb/agent.proto:292]: That's for TLS certificates. Same field is present in all Mongo-related requests.

File: api/managementpb/backup/backups.proto (hunk 1 above)
Comment (artemgavrilov) [reply api/managementpb/backup/backups.proto:190]: I will remove that method for now. There will separate ticket for that.

**Review by fabio-silva — APPROVED** [review]
Seems ok from my side. Just one question for the UI: is every data_model / mode combination ok, or is there any constraint?

**Review by fabio-silva — CHANGES_REQUESTED** [review]
Also, aren't we missing `mode` on `ListScheduled` call?

**Comment by fabio-silva** [comment]
One last question: when we support MySQL, what's gonna be the `mode` there? Are we gonna use `INCREMENTAL` as a generic term for PITR / Incremental?

**Comment by artemgavrilov** [comment]
> One last question: when we support MySQL, what's gonna be the `mode` there? Are we gonna use `INCREMENTAL` as a generic term for PITR / Incremental?

Yes, I think so.

**Review by Dasio — DISMISSED** [review]

**Review by fabio-silva — DISMISSED** [review]

File: api/managementpb/backup/common.proto [hunk 3]
```diff
@@ -10,3 +10,10 @@ enum DataModel {
   PHYSICAL = 1;
   LOGICAL = 2;
 }
+
+// BackupMode specifies backup mode.
+enum BackupMode {
+  BACKUP_MODE_INVALID = 0;
+  SNAPSHOT = 1;
+  INCREMENTAL = 2;
```
Comment (atymchuk) [inline api/managementpb/backup/common.proto:18]: Can we add PITR here as well?
The thing is, SQL engines differentiate between `INCREMENTAL` and `PITR`, so it is by far not the same. If that is true, it'd make sense to have them all here.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: api/managementpb/backup/common.proto (hunk 3 above)
Comment (artemgavrilov) [reply api/managementpb/backup/common.proto:18]: I thought about it, but then checked `pbm` documentation and discovered that there PITR also called "Incremental backups"

https://www.percona.com/doc/percona-backup-mongodb/point-in-time-recovery.html#incremental-backups

Comment (artemgavrilov) [reply api/managementpb/backup/common.proto:18]: On the other hand, it seems that for PG it's different things: https://pgdash.io/blog/postgres-incremental-backup-recovery.html


So yep, you're right. Will add.

Comment (artemgavrilov) [reply api/managementpb/backup/common.proto:18]: Done

**Review by atymchuk — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Review by Dasio — APPROVED** [review]

**Review by oter — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by oter — DISMISSED** [review]

**Review by BupycHuk — DISMISSED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**